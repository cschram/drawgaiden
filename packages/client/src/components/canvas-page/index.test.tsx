import React from 'react';
import renderer from 'react-test-renderer';
import { CanvasPage } from './index';
import testData from '../../../test/data/canvas';

test('<App/>', () => {
    const props = {
        canvasID: testData.canvas.id,
        canvas: testData.canvas,
        history: testData.history,
        latestEntry: testData.newEntry,
        users: testData.users,
        loading: false,
        username: 'username',
        joinCanvas: (id) => {},
        draw: (entry) => {},
        setMousePosition: (coord) => {}
    };
    const tree = renderer.create(<CanvasPage {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
});