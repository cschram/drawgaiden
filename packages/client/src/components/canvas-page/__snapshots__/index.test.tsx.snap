// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> 1`] = `
<div
  className="canvas-page"
>
  <div
    className="canvas-page__easel"
  />
  <div
    className="canvas-page__controls"
  >
    <button
      className="canvas-page__control-toggle btn"
      disabled={false}
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-eye"
      />
    </button>
    <span>
      <ul
        className="radio-button canvas-page__control"
      >
        <li
          className="radio-button__option radio-button__option--checked"
        >
          <input
            checked={true}
            id="tools_0"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="pencil"
          />
          <label
            htmlFor="tools_0"
          >
            <i
              aria-hidden="true"
              className="fa fa-pencil"
            />
          </label>
        </li>
        <li
          className="radio-button__option"
        >
          <input
            checked={false}
            id="tools_1"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="rectangle"
          />
          <label
            htmlFor="tools_1"
          >
            <i
              aria-hidden="true"
              className="fa fa-square-o"
            />
          </label>
        </li>
        <li
          className="radio-button__option"
        >
          <input
            checked={false}
            id="tools_2"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="circle"
          />
          <label
            htmlFor="tools_2"
          >
            <i
              aria-hidden="true"
              className="fa fa-circle-o"
            />
          </label>
        </li>
        <li
          className="radio-button__option"
        >
          <input
            checked={false}
            id="tools_3"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="eraser"
          />
          <label
            htmlFor="tools_3"
          >
            <i
              aria-hidden="true"
              className="fa fa-eraser"
            />
          </label>
        </li>
        <li
          className="radio-button__option"
        >
          <input
            checked={false}
            id="tools_4"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="move"
          />
          <label
            htmlFor="tools_4"
          >
            <i
              aria-hidden="true"
              className="fa fa-arrows-alt"
            />
          </label>
        </li>
        <li
          className="radio-button__option"
        >
          <input
            checked={false}
            id="tools_5"
            name="tools"
            onChange={[Function]}
            type="radio"
            value="colorpicker"
          />
          <label
            htmlFor="tools_5"
          >
            <i
              aria-hidden="true"
              className="fa fa-eyedropper"
            />
          </label>
        </li>
      </ul>
      <div
        className="select canvas-page__control"
        onClick={[Function]}
      >
        <span
          className="select__label"
        >
          Layer 1
        </span>
        <span
          className="select__arrow"
        >
          <i
            aria-hidden="true"
            className="fa fa-caret-down"
          />
        </span>
        <ul
          className="select__menu"
        >
          <li
            className="select__menu-option select__menu-option--active"
            data-value="0"
            onClick={[Function]}
          >
            Layer 1
          </li>
          <li
            className="select__menu-option"
            data-value="1"
            onClick={[Function]}
          >
            Layer 2
          </li>
          <li
            className="select__menu-option"
            data-value="2"
            onClick={[Function]}
          >
            Layer 3
          </li>
        </ul>
      </div>
      <button
        className="canvas-page__control btn"
        disabled={false}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-share-alt"
        />
        <span>
          Share
        </span>
      </button>
      <button
        className="canvas-page__control btn"
        disabled={false}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-floppy-o"
        />
        <span>
          Save
        </span>
      </button>
      <button
        className="canvas-page__control btn"
        disabled={false}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-times"
        />
        <span>
          Exit
        </span>
      </button>
    </span>
  </div>
  <div
    className="canvas-page__settings"
  >
    <div
      className="canvas-page__tool-setting canvas-page__colors"
    >
      <div
        className="color-picker"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
          }
        }
      >
        <input
          onChange={[Function]}
          type="color"
          value="#000000"
        />
      </div>
      <div
        className="color-picker"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <input
          onChange={[Function]}
          type="color"
          value="#ffffff"
        />
      </div>
    </div>
    <div
      className="canvas-page__tool-setting"
    >
      <label>
        Size
      </label>
      <input
        className="slider "
        disabled={false}
        max={50}
        min={1}
        name="size"
        onChange={[Function]}
        step={1}
        type="range"
        value={1}
      />
    </div>
    <div
      className="canvas-page__tool-setting"
    >
      <label>
        Opacity
      </label>
      <input
        className="slider "
        disabled={false}
        max={100}
        min={0}
        name="opacity"
        onChange={[Function]}
        step={1}
        type="range"
        value={100}
      />
    </div>
    <div
      className="canvas-page__tool-setting"
    >
      <label>
        Smoothness
      </label>
      <input
        className="slider "
        disabled={false}
        max={100}
        min={0}
        name="smoothness"
        onChange={[Function]}
        step={1}
        type="range"
        value={80}
      />
    </div>
  </div>
</div>
`;
